xbmc/pvr/PVRGUIInfo.cpp

```diff
@@ -288,15 +298,17 @@ void CPVRGUIInfo::UpdateTimeshift(void)
 
   if (m_iTimeshiftEndTime > m_iTimeshiftStartTime)
   {
-    // getstreamtimes api
+    // timeshifting supported
     m_iTimeshiftPlayTime = iStartTime + iPlayTime;
   }
-  else
+  else if (bPlaying)
   {
-    // legacy api
-    m_iTimeshiftPlayTime = std::time(nullptr);
+    // timeshifting not supported
+    m_iTimeshiftPlayTime = now - m_iTimeshiftOffset;
   }
 
+  m_iTimeshiftOffset = now - m_iTimeshiftPlayTime;
```

Comment:
@ksooo system time of kodi box is not synchronized with backend or wall clock. means "now" can be smaller than m_iTimeshiftPlayTime.
We should use buffer end time for "now"